### ISSUE TITLE
add some UX for "restart kernel" 

### ISSUE THREAD

**Issue by bcolloran** [iodide-project/iodide#998]
see #994 for one use case, though this will be useful more generally -- idea is for users to have a way to restart the iframe, and also to detect when something happens that causes the connection to the iframe to be lost for some reason

**Comment by bcolloran** [comment]
related to #971 (in that this will a require a more robust way of having the editor and eval-frame listen for each other and connect to each other), so a solution to this might (optionally) try to address that as well)

**Comment by bcolloran** [comment]
(if you want to try to address this and #971 in  one go, feel free to unassign me from that one)

**Comment by mdboom** [comment]
I'm thinking of deferring this until after #1106 lands -- the UX will be *very* related to that status -- so much so that I think it might make sense to make it the same thing.  What do you think, @hamilton ?

**Comment by hamilton** [comment]
@mdboom I agree. Let's assign this to me and I can tackle it in a subsequent PR.
